### DISCUSSION TITLE
cannot access proxy on Synology

### DISCUSSION THREAD

**Discussion by shawn8888** [haugene/docker-transmission-openvpn#2491] [Q&A]
I setup the docker on a Synology server and transmission works great.
However I don't know why the proxy doesn't work. Here is my setup:

Ports:

![image](https://user-images.githubusercontent.com/8271101/211221284-15cf384e-6c44-41bf-958c-933f53851c40.png)
![image](https://user-images.githubusercontent.com/8271101/211221396-e3242ca2-4837-4c16-b124-0df50f3406f2.png)

Log also shows that Privoxy is running on 8118 port successfully.
![image](https://user-images.githubusercontent.com/8271101/211221766-6d91a495-9da0-4571-8ebc-5deca5ab1734.png)

But proxy doesn't work for my browser (192.168.100.155 is my Synology server IP):
![image](https://user-images.githubusercontent.com/8271101/211221785-7b10cdae-911a-421b-bd83-47b4a4b6f587.png)

Am I doing anything wrong?

**Comment by shawn8888** [comment]
Here is the exported settings
docker runs on Synology DSM 6
[PIA.Settings.txt](https://github.com/haugene/docker-transmission-openvpn/files/10376869/PIA.Settings.txt)

**Comment by shawn8888** [comment]
Update:
1. I checked the config file in the container:
 /etc/privoxy/config
It shows
`listen-address 172.21.0.2:8888`
regardless I set the port to 8118 and the log also shows the port is 8118.
2. I changed WEBPROXY_PORT to 8888 and add the port mapping 8888 to 8888 (local port to container port)
3. Set browser proxy to NAS-IP:8888. Test it and it works!
4. The tutorial says the web ui is on config.privoxy.org 
![image](https://user-images.githubusercontent.com/8271101/211456808-fa7f894a-ac52-4ee3-a773-9d2856dc0000.png)
However, I found that 
http://config.privoxy.org/   isn't working and shows "Privoxy is not being used" error.
But http://p.p/ is working.
Hope above helps for others. 

My biggest question is, why WEBPROXY_PORT doesn't work? Is it a bug?

**Reply by Dylan-Bs** [reply]
Yes, I thinks you're right, WEBPROXY_PORT don't work. 

For me, it's set too 8888 since 2 years, and now I noticed it stopped to work because it's 8118 in the config file ... 

Also, I found in my logs "2023-01-23T17:55:27.611390239Z Privoxy: ERROR. Supplied port =8118 is not a number
2023-01-23T17:55:27.611677353Z Privoxy: Using config file at /etc/privoxy/config
"

Thanks for your help !

**Comment by edgd1er** [comment]
Hello,

* is your image the latest ?
* You may try to run the container with DEBUG=true to have more information.
* http://config.privoxy.org/show-status was available in the browser but I had to refresh several times the page as I ended 3 or 4 time to the wrong page.

As your are using bridge network, defining webproxy_port has no real use as exposed port is mapped to the guest port. I suppose this setting is useful when network is set to host or macvlan

**Reply by shawn8888** [reply]
1. How do I tell the image's version? I did update not long ago.
I set DEBUG=true, and the only version information I see is:
`2023-01-10 08:20:34 library versions: OpenSSL 3.0.2 15 Mar 2022, LZO 2.10`

2.
http://config.privoxy.org/  not working, but
http://config.privoxy.org/show-status  works!
However, when I click the links such as "listen-address", it gives me "500 Internal Privoxy Error"
Does that mean I can only change config by editing the config file?

3.
I know little about networking in docker. You are right I can map any ports from local to container by bridge mode. But still the WEBPROXY_PORT is misleading in the HELP documents.
